bridge-impl/src/main/java/com/liferay/faces/bridge/scope/internal/BridgeRequestScopeManagerImpl.java

```diff
@@ -39,105 +43,164 @@
 	// Logger
 	private static final Logger logger = LoggerFactory.getLogger(BridgeRequestScopeManagerImpl.class);
 
-	public void removeBridgeRequestScopesByPortlet(PortletConfig portletConfig) {
-		String portletNameToRemove = portletConfig.getPortletName();
-		PortletContext portletContext = portletConfig.getPortletContext();
-		BridgeRequestScopeCache bridgeRequestScopeCache = BridgeRequestScopeCacheFactory
-			.getBridgeRequestScopeCacheInstance(portletContext);
-		Set<Map.Entry<String, BridgeRequestScope>> mapEntries = bridgeRequestScopeCache.entrySet();
+	// Private Constants
+	private static final String ATTR_BRIDGE_REQUEST_SCOPE_CACHE = "com.liferay.faces.bridge.bridgeRequestScopeCache";
+
+	@Override
+	public Cache<String, BridgeRequestScope> getBridgeRequestScopeCache(PortletContext portletContext) {
+
+		Cache<String, BridgeRequestScope> bridgeRequestScopeCache;
+
+		synchronized (portletContext) {
+
+			bridgeRequestScopeCache = (Cache<String, BridgeRequestScope>) portletContext.getAttribute(
+					ATTR_BRIDGE_REQUEST_SCOPE_CACHE);
 
-		if (mapEntries != null) {
+			if (bridgeRequestScopeCache == null) {
 
-			List<String> keysToRemove = new ArrayList<String>();
+				CacheFactory cacheFactory = (CacheFactory) BridgeFactoryFinder.getFactory(portletContext,
+						CacheFactory.class);
 
-			for (Map.Entry<String, BridgeRequestScope> mapEntry : mapEntries) {
-				BridgeRequestScope bridgeRequestScope = mapEntry.getValue();
-				String bridgeRequestScopeId = bridgeRequestScope.getId();
-				String portletName = bridgeRequestScopeId.split("[:][:][:]")[0];
+				// Spec Section 3.2: Support for configuration of maximum number of bridge request scopes.
+				Integer maxManagedRequestScope = null;
+				String maxManagedRequestScopeString = portletContext.getInitParameter(
+						Bridge.MAX_MANAGED_REQUEST_SCOPES);
```

Comment:
@ngriffin7a we cannot use `PortletConfigParam` since it is designed to use `portletConfig` and not `portletContext` (this is the only use case where that happens so I don't think it's worth adding that functionality to `PortletConfigParam`).